File: csharp/main track/05 resistor-color/ResistorColor.cs
﻿using System;

public static class ResistorColor
{
    private static readonly string[] resistorColors = { "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" };
    public static int ColorCode(string color) => Array.IndexOf(resistorColors, color);
    public static string[] Colors() => resistorColors;
}

File: java/side exercises/resistor-color/src/main/java/ResistorColor.java
import java.util.List;
import java.util.ArrayList;

class ResistorColor {
	
	List<String> resistors = new ArrayList<>();
	
	ResistorColor () {
		resistors.add ("black");
		resistors.add ("brown");
		resistors.add ("red");
		resistors.add ("orange");
		resistors.add ("yellow");
		resistors.add ("green");
		resistors.add ("blue");
		resistors.add ("violet");
		resistors.add ("grey");
		resistors.add ("white");
	}
	
    int colorCode(String color) {
        		
		return resistors.indexOf(color);
    }

    String[] colors() {
        return resistors.toArray(new String[resistors.size()]);
    }
}
